File: lessons/day2/04-vscode-ide/exercises/broken_script.py
"""broken_script.py — a deliberately messy script for the linting exercise.

Save the file in VS Code with ruff configured. Watch the diagnostics appear.
Some are auto-fixable; some are real bugs you must read and decide on.

This file MUST stay broken. Do not "fix" it before the workshop.
"""

import os
import sys
import json   # unused — ruff F401
from pathlib import Path
import pandas as pd


def parse_samplesheet(path):
    df = pd.read_csv(path)
    return df


def summarize(df):
    # Bug: `n` shadows the loop variable below. ruff will flag it (PLW2901-ish
    # via `--select PL`) and a careful reader catches it because `n` ends up
    # being whatever the last row's read_count was, not the total.
    n = 0
    for n in df["read_count"]:
        n + n   # noqa — wrong intent: the writer meant `total += n`
    return n


def main(argv):
    samplesheet = argv[1]
    df = parse_samplesheet(samplesheet)
    total = summarize(df)
    print(f"total reads across {len(df)} samples: {total}")


if __name__ == "__main__":
    main(sys.argv)

File: lessons/day2/04-vscode-ide/exercises/broken_script.R
# broken_script.R — a deliberately messy R script for the linting exercise.
#
# Open in VS Code with the R extension and lintr enabled. Save. The diagnostics
# appear. Some are stylistic; some are real bugs.

library(dplyr)
library(readr)
library(stringr)  # unused — lintr unused_import_linter

# Bug: assignment with = inside a function call boundary.
parse_samplesheet = function(path){
  df<-read_csv(path,show_col_types=FALSE)
  return(df)
}

# Bug: variable name `T` shadows TRUE.
summarize_reads = function(df){
  T = 0
  for (n in df$read_count){
    T = T+n
  }
  return(T)
}

main <- function(argv) {
  ss <- argv[1]
  df <- parse_samplesheet(ss)
  total <- summarize_reads(df)
  cat(sprintf("total reads across %d samples: %d\n", nrow(df), total))
}

args <- commandArgs(trailingOnly = TRUE)
main(args)
